src/advanced_inv.cpp

```diff
@@ -1752,15 +1753,18 @@ bool advanced_inventory::remove_item( advanced_inv_listitem &sitem )
         assert( &cont->contents.front() == sitem.it );
         cont->contents.erase( cont->contents.begin() );
         rc = true;
-    } else if( s.can_store_in_vehicle() && panes[src].in_vehicle() ) {
-        rc = s.veh->remove_item( s.vstor, sitem.it );
     } else if( sitem.area == AIM_WORN ) {
         rc = g->u.takeoff( sitem.it );
-    } else {
+    } else if( s.can_store_in_vehicle() ) {
+        rc = s.veh->remove_item( s.vstor, sitem.it );
+    }
+
+    if( !rc ) {
         g->m.i_rem( s.pos, sitem.it );
-        rc = true;
+        rc = true; // NOT ALWAYS! FIX!
     }
-    return rc;
+
+    return rc; // Always returns true
```

Comment:
> Always returns true

Than why make it a variable? Variable comes from good old Latin variare which apparently means “to change”, but if it's always the same, there is no change (-:

That said, the function should not return anything. It's already to late to react! The caller *must* make sure that removing the item is possible. Coincidentally, `map::i_rem` does not return anything either.